src/main/java/com/avispl/symphony/dal/avdevices/encoderdecoder/haivision/xdecoder/HaivisionXDecoderCommunicator.java

```diff
@@ -696,14 +819,803 @@ public boolean validatePortRange(Integer portNumber) {
+				String enableSyncMode = SyncMode.ENABLE_SYNC_MODE.getName();
+				if (!isEnableSyncMode) {
+					enableSyncMode = SyncMode.DISABLE_SYNC_MODE.getName();
+				}
+				cachedDecoderConfig.setEnableBuffering(enableSyncMode);
+				this.cachedDecoderConfigs.set(decoderID, cachedDecoderConfig);
+				addAdvanceControlProperties(advancedControllableProperties, createSwitch(stats, decoderControllingGroup + DecoderControllingMetric.SYNC_MODE.getName(), Integer.parseInt(value),
+						DecoderConstant.DISABLE, DecoderConstant.ENABLE));
+				populateApplyChangeAndCancelButtonForDecoder(stats, advancedControllableProperties, decoderID);
+				populateLocalExtendedStats(stats, advancedControllableProperties);
+				break;
+			case BUFFERING_MODE:
+				BufferingMode bufferingMode = BufferingMode.getByUiName(value);
+				cachedDecoderConfig.setBufferingMode(bufferingMode.getApiName());
+				this.cachedDecoderConfigs.set(decoderID, cachedDecoderConfig);
+				populateDecoderControlBufferingMode(stats, advancedControllableProperties, cachedDecoderConfig, decoderID);
+				populateApplyChangeAndCancelButtonForDecoder(stats, advancedControllableProperties, decoderID);
+				populateLocalExtendedStats(stats, advancedControllableProperties);
+				break;
+			case BUFFERING_DELAY:
+				Integer bufferingDelay = DecoderConstant.MIN_BUFFERING_DELAY;
+				try {
+					bufferingDelay = Integer.parseInt(value);
+					if (bufferingDelay < DecoderConstant.MIN_BUFFERING_DELAY) {
+						bufferingDelay = DecoderConstant.MIN_BUFFERING_DELAY;
+					}
+					if (bufferingDelay > DecoderConstant.MAX_BUFFERING_DELAY) {
+						bufferingDelay = DecoderConstant.MAX_BUFFERING_DELAY;
+					}
+				} catch (Exception e) {
+					if (logger.isWarnEnabled()) {
+						logger.warn("Invalid buffering delay value");
+					}
+				}
+				cachedDecoderConfig.setBufferingDelay(value);
+				this.cachedDecoderConfigs.set(decoderID, cachedDecoderConfig);
+				advancedControllableProperties.add(createNumeric(stats, decoderControllingGroup + DecoderControllingMetric.BUFFERING_DELAY.getName(), bufferingDelay.toString()));
+				populateApplyChangeAndCancelButtonForDecoder(stats, advancedControllableProperties, decoderID);
+				populateLocalExtendedStats(stats, advancedControllableProperties);
+				break;
+			case MULTI_SYNC_BUFFERING_DELAY:
+				Integer multiSyncBufferingDelay = DecoderConstant.DEFAULT_MULTI_SYNC_BUFFERING_DELAY;
+				try {
+					multiSyncBufferingDelay = Integer.parseInt(value);
+					if (multiSyncBufferingDelay < DecoderConstant.MIN_MULTI_SYNC_BUFFERING_DELAY) {
+						multiSyncBufferingDelay = DecoderConstant.MIN_MULTI_SYNC_BUFFERING_DELAY;
+					}
+					if (multiSyncBufferingDelay > DecoderConstant.MAX_MULTI_SYNC_BUFFERING_DELAY) {
+						multiSyncBufferingDelay = DecoderConstant.MAX_MULTI_SYNC_BUFFERING_DELAY;
+					}
+				} catch (Exception e) {
+					if (logger.isWarnEnabled()) {
+						logger.warn("Invalid buffering delay value");
```

Comment:
Also log for catched exception.
logger.warn("Invalid buffering delay value",e);

Reply:
I fixed it